bee-api/bee-rest-api/src/endpoints/routes/api/v1/messages/mod.rs

```diff
@@ -0,0 +1,152 @@
+use crate::AppStorage;
+use crate::types::{
+    dtos::MessageDto,
+    body::{SuccessBody, ErrorBody, DefaultErrorResponse},
+    responses::{SubmitMessageResponse, MessageResponse, MessageMetadataResponse, MessageChildrenResponse},
+};
+
+use bee_message::{Message, MessageId};
+use bee_storage::access::{Fetch, Insert};
+
+use axum::{
+    extract::{Path, Extension},
+    routing::{get, post},
+    response::Json,
+    Router,
+};
+use serde_json::{Value, json};
+
+use std::{
+    sync::Arc,
+    str::FromStr,
+};
+use std::convert::{TryFrom};
+use rand::Rng;
+
+pub fn api_routes() -> Router {
+    Router::new()
+        .route("/", post(post_handler))
+        .route("/:messageId", get(get_id_handler))
+        .route("/:messageId/metadata", get(get_id_metadata_handler))
+        .route("/:messageId/raw", get(get_id_raw_handler))
+        .route("/:messageId/children", get(get_id_children_handler))
+}
+
+pub async fn post_handler(Extension(app_storage): Extension<Arc<AppStorage>>, Json(payload): Json<MessageDto>) -> Json<Value> {
+    let mut rng = rand::thread_rng();
+
+    let rand_part1: u128 = rng.gen();
+    let rand_part2: u128 = rng.gen();
+    let rand_id = format!("{:x}{:x}", rand_part1, rand_part2);
+    match Insert::<MessageId, Message>::insert(&*(
+        app_storage.storage.lock().unwrap()),
```

Comment:
I have 2 comments about this part:
- Is there any reason to not use an `async` oriented lock here?
- The `unwrap` should have a comment explaining why it cannot fail.